### ISSUE TITLE
Add Rule: no-disconnected-chained-awaits

### ISSUE THREAD

**Issue by Nokel81** [eslint/eslint#10130]
**Please describe what the rule should do:**
This rule should warn when there are awaits of function calls (including news) next to each other where the awaits do not depend of the values of the previous awaits.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ ] Warns about a potential error
[ ] Suggests an alternate way of doing something
[x] Other (please specify:)
If this is the case then the code will run slower since it forces the code to run sequentially when it could run more parallelly. 

**Provide 2-3 code examples that this rule will warn about:**

This would warn:
```js
async function fn() {
    await gn();
    await hn();
    return true;
}
```
This would warn:
```js
async function fn() {
    const a = await gn();
    const b = await hn();
    return {a, b};
}
```
This would not warn:
```js
async function fn() {
    const a = await gn();
    const b = await hn(a);
    return b;
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because with the move towards async/await and promises over callbacks this will be a common mistake for new people. It is also a good catch for experienced developers.

**Comment by Nokel81** [comment]
The first step would probably be just the warning as the `--fix` might be considered too intrusive or maybe even too difficult.

**Comment by Nokel81** [comment]
First Example:
```js
async function fn() {
    await gn();
    await hn();
    return true;
}
```
Could be rewritten as the following:
```js
async function fn() {
    const gn_promise = gn();
    const hn_promise = hn();
    await gn_promise;
    await hn_promise;
    return true;
}
```

**Comment by Nokel81** [comment]
Second Example:
```js
async function fn() {
    const a = await gn();
    const b = await hn();
    return {a, b};
}
```
Could be rewritten as the following:
```js
async function fn() {
    const gn_promise = gn();
    const hn_promise = hn();
    const a = await gn_promise;
    const b = await hn_promise;
    return {a, b};
}
```

**Comment by Nokel81** [comment]
The reason why this is the correct why to rewrite them in pure ecmascript is that an await is a blocking call on the promise. Meaning that if a chain of these are in a row then each long action will be done serially. 

However, if they are written as the second snippets then both long actions will be sent away and both can complete together

**Comment by BridgeAR** [comment]
I personally like the proposal but some applications might actually need to verify that async operation a is done before async operation b even though they seem to be independent.

About rewriting them: I would suggest to use `Promise.all` in these cases as that would be less code, e.g.:

```js
async function fn() {
  const [a, b] = await Promise.all([gn(), hn()]);
  return {a, b};
}
```

**Comment by ljharb** [comment]
Promise.all would be better, for sure.

**Comment by Nokel81** [comment]
That makes sense to me, I don't think that this should ever be a auto fix

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is a strong consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
